aineko/extras/connectors/wss.py

```diff
@@ -0,0 +1,209 @@
+# Copyright 2023 Aineko Authors
+# SPDX-License-Identifier: Apache-2.0
+"""Extra module for connecting to a WebSocket."""
+
+import json
+import time
+from typing import Any, Dict, List, Optional
+
+import websocket
+from pydantic import BaseModel, field_validator
+
+from aineko import AbstractNode
+from aineko.utils.secrets import inject_secrets
+
+
+class ParamsWSS(BaseModel):
+    """Connector params for WebSocket model."""
+
+    max_retries: int = 30
+    retry_sleep: float = 5
+    url: str
+    header: Optional[Dict[str, str]] = None
+    init_messages: List[Any] = []
+    metadata: Optional[Dict[str, Any]] = None
+
+    @field_validator("url")
+    def supported_url(cls, u: str) -> str:  # pylint: disable=no-self-argument
+        """Validates that the url is a valid WebSocket URL."""
+        if not (u.startswith("wss://") or u.startswith("ws://")):
+            raise ValueError(
+                "Invalid url provided to WebSocket params. "
+                'Expected url to start with "wss://" or "ws://". '
+                f"Provided url was: {str(u)}"
```

Comment:
```suggestion
                f"Provided url was: {u}"
```
The str() is redundant when using f-strings.

I noticed this pattern in the rest of the code as well, but I'll only comment on this instance fyi.

Reply:
oh i mis-interpreted a mypy error i saw. fixed here: https://github.com/aineko-dev/aineko/pull/121/commits/d71cd6b4c6b3f85bd4230ef93074e42ece1fdf78